<a href="https://colab.research.google.com/github/toan27062002/baitap/blob/main/bai_tap_phan_tich_anova_1_chieu.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Hãy sử dụng các tập dữ liệu sau và tiến hành phân tích ANOVA 1 chiều

- [Excavation Depth and Archaeology](https://college.cengage.com/mathematics/brase/understandable_statistics/7e/students/datasets/owan/frames/owan01.html)
- [Apple Orchard Experiment](https://college.cengage.com/mathematics/brase/understandable_statistics/7e/students/datasets/owan/frames/owan02.html)
- [Red Dye Number 40](https://college.cengage.com/mathematics/brase/understandable_statistics/7e/students/datasets/owan/frames/owan03.html)
- [Business Startup Costs](https://college.cengage.com/mathematics/brase/understandable_statistics/7e/students/datasets/owan/frames/owan04.html)
- [Weights of Football Players](https://college.cengage.com/mathematics/brase/understandable_statistics/7e/students/datasets/owan/frames/owan05.html)

In [188]:
#Code here

In [189]:
import numpy as np
import pandas as pd
from scipy import stats as st
from scipy.stats import f_oneway

In [190]:
# kiểm định phân phối chuẩn 
def shapiro_test(sample):
    statistic, p_value = st.shapiro(sample)
    if (statistic > 0) and (p_value > .05):
        print('Mẫu tuân theo phân phối chuẩn.')
    else:
        print('Mẫu không tuân theo phân phối chuẩn.')

In [191]:
# kiểm định phương sai
def bartlett_test_4(c1, c2, c3, c4):
    bartlett_test = st.bartlett(c1, c2, c3, c4)
    bartlettvalue,pvalue = bartlett_test
    if (bartlettvalue > 0) and (pvalue > .05):
        print('Phương sai các nhóm tương đồng.')
    else:
        print('Phương sai các nhóm không không tương đồng.')   

def bartlett_test_5(c1, c2, c3, c4, c5):
    bartlett_test = st.bartlett(c1, c2, c3, c4, c5)
    bartlettvalue,pvalue=bartlett_test
    if (bartlettvalue > 0) and (pvalue > .05):
        print('Phương sai các nhóm bằng nhau.')
    else:
        print('Phương sai các nhóm không bằng nhau.')

In [192]:
# kiểm định anova
def anova_4(g1, g2, g3, g4):
    statistic,p_value = f_oneway(g1, g2, g3, g4)
    print('statistic:',statistic, '; p value:',p_value)    
    if p_value < .05:
        print("Có bằng chứng để bác bỏ giả thuyết H0")
    else:
        print("Chưa có bằng chứng để bác bỏ giả thuyết H0")

def anova_5(g1, g2, g3, g4, g5):
    statistic,p_value = f_oneway(g1, g2, g3, g4, g5)
    print('statistic:',statistic, '; p value:',p_value)    
    if p_value < .05:
        print("Có bằng chứng để bác bỏ giả thuyết H0")
    else:
        print("Chưa có bằng chứng để bác bỏ giả thuyết H0")

câu 1:

In [193]:
df1 = pd.read_excel('owan01.xls')
df1.head()

*** No CODEPAGE record, no encoding_override: will use 'ascii'


X1    X2     X3    X4
0   93  85.0  100.0  96.0
1  120  45.0   75.0  58.0
2   65  80.0   65.0  95.0
3  105  28.0   40.0  90.0
4  115  75.0   73.0  65.0

In [194]:
x1_1 = df1['X1']
x1_2 = df1['X2']
x1_3 = df1['X3']
x1_4 = df1['X4']

In [195]:
# kiểm định phân phối chuẩn
shapiro_test(x1_1)
shapiro_test(x1_2)
shapiro_test(x1_3)
shapiro_test(x1_4)

Mẫu tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.


In [196]:
# vì có cột dữ liệu không tuân theo phân phối chuẩn nên k thể phân tích anova

câu 2 :

In [197]:
df2 = pd.read_excel('owan02.xls')
df2.head()

*** No CODEPAGE record, no encoding_override: will use 'ascii'


X1    X2    X3    X4    X5
0  2569  2074  2505  2838  1532
1  2928  2885  2315  2351  2552
2  2865  3378  2667  3001  3083
3  3844  3906  2390  2439  2330
4  3027  2782  3021  2199  2079

In [198]:
x2_1 = df2['X1']
x2_2 = df2['X2']
x2_3 = df2['X3']
x2_4 = df2['X4']
x2_5 = df2['X5']

In [199]:
# kiểm định phân phối chuẩn
shapiro_test(x2_1)
shapiro_test(x2_2)
shapiro_test(x2_3)
shapiro_test(x2_4)
shapiro_test(x2_5)

Mẫu tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.


In [200]:
bartlett_test_5(x2_1, x2_2, x2_3, x2_4, x2_5 )

Phương sai các nhóm bằng nhau.


In [201]:
anova_5(x2_1, x2_2, x2_3, x2_4, x2_5)

statistic: 1.3142198672637213 ; p value: 0.28382583646160114
Chưa có bằng chứng để bác bỏ giả thuyết H0


ANOVA một chiều sử dụng giả thuyết rỗng và giả thuyết thay thế sau:

$$H_{0}: \mu{1} = \mu{2} =\mu{3} =\mu{4} =\mu{5} $$ (tất cả các trung bình tổng thể đều bằng nhau)
$$H{a} :$$ ít nhất một trung bình của tổng thể khác  với phần còn lại

Thống kê kiểm định F là  1.314 và giá trị p tương ứng là 0.283 . Vì giá trị p không nhỏ hơn 0,05 nên chúng tôi không thể bác bỏ giả thuyết không. Điều này có nghĩa là chúng tôi không có đủ bằng chứng để nói rằng có sự khác biệt về sự tăng trưởng mở rộng giữa 5 loại rễ.

Câu 3:

In [202]:
df3 = pd.read_excel('owan03.xls')
df3.head()

*** No CODEPAGE record, no encoding_override: will use 'ascii'


X1    X2    X3    X4
0  70  49.0  30.0  34.0
1  77  60.0  37.0  36.0
2  83  63.0  56.0  48.0
3  87  67.0  65.0  48.0
4  92  70.0  76.0  65.0

In [203]:
x3_1 = df3['X1']
x3_2 = df3['X2']
x3_3 = df3['X3']
x3_4 = df3['X4']

In [204]:
# kiểm định phân phối chuẩn
shapiro_test(x3_1)
shapiro_test(x3_2)
shapiro_test(x3_3)
shapiro_test(x3_4)

Mẫu tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.


In [205]:
# vì có cột dữ liệu không tuân theo phân phối chuẩn nên k thể phân tích anova

Câu 4:

In [206]:
df4 = pd.read_excel('owan04.xls')
df4.head()

*** No CODEPAGE record, no encoding_override: will use 'ascii'


X1     X2    X3     X4  X5
0   80.0  150.0  48.0  100.0  25
1  125.0   40.0  35.0   96.0  80
2   35.0  120.0  95.0   35.0  30
3   58.0   75.0  45.0   99.0  35
4  110.0  160.0  75.0   75.0  30

In [207]:
x4_1 = df4['X1']
x4_2 = df4['X2']
x4_3 = df4['X3']
x4_4 = df4['X4']
x4_5 = df4['X5']

In [208]:
# kiểm định phân phối chuẩn
shapiro_test(x4_1)
shapiro_test(x4_2)
shapiro_test(x4_3)
shapiro_test(x4_4)
shapiro_test(x4_5)

Mẫu không tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.
Mẫu không tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.


In [209]:
# vì có cột dữ liệu không tuân theo phân phối chuẩn nên k thể phân tích anova

Câu 5:


In [210]:
df5 = pd.read_excel('owan05.xls')
df5.head()

*** No CODEPAGE record, no encoding_override: will use 'ascii'


X1   X2   X3   X4   X5
0  250  260  270  260  247
1  255  271  250  255  249
2  255  258  281  265  255
3  264  263  273  257  247
4  250  267  257  268  244

In [211]:
x5_1 = df5['X1']
x5_2 = df5['X2']
x5_3 = df5['X3']
x5_4 = df5['X4']
x5_5 = df5['X5']

In [212]:
# kiểm định phân phối chuẩn
shapiro_test(x5_1)
shapiro_test(x5_2)
shapiro_test(x5_3)
shapiro_test(x5_4)
shapiro_test(x5_5)

Mẫu không tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.
Mẫu tuân theo phân phối chuẩn.
